From 4a9292f16915345b5a02a43eaa244d865d9ea49c Mon Sep 17 00:00:00 2001
From: Andy Shaw <andy.shaw@qt.io>
Date: Thu, 05 Sep 2019 12:52:43 +0200
Subject: [PATCH] Ensure all children of a widget get updated when a stylesheet changes

Before, only the direct children would get an update when the
stylesheet changed, any children below that would be unchanged.

Fixes: QTBUG-77006
Change-Id: Id668eaae74a8289d78d66644f077e6a3302960cd
Reviewed-by: Frederik Gladhorn <frederik.gladhorn@qt.io>
(cherry picked from commit 9b6179cf957c32e0c02547d510dfee2088f02340)
---

diff --git a/src/widgets/styles/qstylesheetstyle.cpp b/src/widgets/styles/qstylesheetstyle.cpp
index 28b4c36..2b5db65 100644
--- a/src/widgets/styles/qstylesheetstyle.cpp
+++ b/src/widgets/styles/qstylesheetstyle.cpp
@@ -2732,6 +2732,11 @@
         if (auto widget = qobject_cast<QWidget*>(const_cast<QObject*>(object))) {
             widget->style()->polish(widget);
             QApplication::sendEvent(widget, &event);
+            QList<const QObject *> children;
+            children.reserve(widget->children().size() + 1);
+            for (auto child: qAsConst(widget->children()))
+                children.append(child);
+            updateObjects(children);
         }
     }
 }
